src/BenchmarkDotNet.TestAdapter/BenchmarkCaseExtensions.cs

```diff
@@ -0,0 +1,94 @@
+﻿using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Characteristics;
+using BenchmarkDotNet.Exporters;
+using BenchmarkDotNet.Extensions;
+using BenchmarkDotNet.Running;
+using Microsoft.TestPlatform.AdapterUtilities;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using System;
+
+namespace BenchmarkDotNet.TestAdapter
+{
+    /// <summary>
+    /// A set of extensions for BenchmarkCase to support converting to VSTest TestCase objects.
+    /// </summary>
+    internal static class BenchmarkCaseExtensions
+    {
+        /// <summary>
+        /// Converts a BDN BenchmarkCase to a VSTest TestCase.
+        /// </summary>
+        /// <param name="benchmarkCase">The BenchmarkCase to convert.</param>
+        /// <param name="assemblyPath">The dll or exe of the benchmark project.</param>
+        /// <param name="includeJobInName">Whether or not the display name should include the job name.</param>
+        /// <returns>The VSTest TestCase.</returns>
+        internal static TestCase ToVSTestCase(this BenchmarkCase benchmarkCase, string assemblyPath, bool includeJobInName=false)
+        {
+            var benchmarkMethod = benchmarkCase.Descriptor.WorkloadMethod;
+            var fullClassName = benchmarkCase.Descriptor.Type.GetCorrectCSharpTypeName();
+            var benchmarkMethodName = benchmarkCase.Descriptor.WorkloadMethod.Name;
+            var benchmarkFullName = $"{fullClassName}.{benchmarkMethodName}";
+
+            // Display name has arguments as well.
+            var displayMethodName = FullNameProvider.GetMethodName(benchmarkCase);
+            if (includeJobInName)
+                displayMethodName += $" [{benchmarkCase.GetUnrandomizedJobDisplayInfo()}]";
+
+            var displayName = $"{fullClassName}.{displayMethodName}";
+
+            var vsTestCase = new TestCase(benchmarkFullName, VSTestAdapter.ExecutorUri, assemblyPath)
```

Comment:
Currently, several test cases may have equal `fullyQualifiedName`. It will be better to have a unique FQN for each test case.

Reply:
`fullyQualfiedName` is a legacy property now and is not intended to be used as a unique identifier for a test, rather the `Id` property is the only thing that needs to uniquely identify a case.

Despite it being legacy, it still exists and is required for the purposes of extracting the project/namespace/class/testgroup hierarchy view that is shown in the VS Test Explorer: https://github.com/microsoft/vstest/blob/main/docs/RFCs/0033-Hierarchy-TestCase-Property.md#overview